x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/inference/assignment/TrainedModelAssignment.java

```diff
@@ -224,15 +224,15 @@ public boolean hasStartedRoutes() {
         return nodeRoutingTable.values().stream().anyMatch(routeInfo -> routeInfo.getState() == RoutingState.STARTED);
     }
 
-    public List<Tuple<String, Integer>> selectRandomStartedNodesWeighedOnAllocationsForNRequests(
+    public List<Tuple<String, Integer>> selectRandomNodesWeighedOnAllocationsForNRequestsAndState(
```

Comment:
I'd just call this `selectRandomNodes(numberOfRequests, acceptableStates)` and put the documentation in JavaDoc instead of in the method name :)